shipyard_blueprints/sftp/shipyard_sftp/cli/delete_file.py

```diff
@@ -29,129 +33,75 @@ def get_args():
+            raise ExitCodeException(
+                "Must specify a password or a RSA key",
+                CloudStorage.EXIT_CODE_INVALID_CREDENTIALS,
+            )
+
+        connection_args = {"host": args.host, "port": args.port, "user": args.username}
+
+        if args.password:
+            connection_args["pwd"] = args.password
+
+        if key := args.key:
+            if not os.path.isfile(key):
+                fd, key_path = tempfile.mkstemp()
+                logger.info(f"Storing RSA temporarily at {key_path}")
+                with os.fdopen(fd, "w") as tmp:
+                    tmp.write(key)
+                connection_args["key"] = key_path
+        sftp = SftpClient(**connection_args)
+
+        source_file_name = args.source_file_name
+        source_folder_name = shipyard.clean_folder_name(args.source_folder_name)
+        source_full_path = shipyard.combine_folder_and_file_name(
+            source_folder_name, source_file_name
         )
-        if len(matching_file_names) == 0:
-            print("No matches found")
-            sys.exit(ec.EXIT_CODE_NO_MATCHES_FOUND)
-        print(f"{len(matching_file_names)} files found. Preparing to delete...")
-
-        for index, file_name in enumerate(matching_file_names, 1):
-            delete_file_path = file_name
-
-            print(f"deleting file {index} of {len(matching_file_names)}")
-            try:
-                delete_sftp_file(client, delete_file_path)
-            except Exception as e:
-                print(f"Failed to delete {file_name}... Skipping")
-    else:
-        delete_sftp_file(client=client, file_path=source_full_path)
-
-    if key_path:
-        print(f"Removing temporary RSA Key file {key_path}")
-        os.remove(key_path)
+        source_file_name_match_type = args.source_file_name_match_type
+
+        if source_file_name_match_type == "regex_match":
+            file_names = sftp.list_files_recursive(source_folder_name or ".")
+            matching_file_names = shipyard.find_all_file_matches(
+                file_names, re.compile(source_file_name)
+            )
+            if len(matching_file_names) == 0:
+                raise ExitCodeException(
+                    "No matches found", sftp.EXIT_CODE_FILE_NOT_FOUND
+                )
+            logger.info(
+                f"{len(matching_file_names)} files found. Preparing to delete..."
+            )
+
+            for index, file_name in enumerate(matching_file_names, 1):
+                delete_file_path = file_name
+
+                logger.info(f"Deleting file {index} of {len(matching_file_names)}")
+                try:
+                    sftp.remove(delete_file_path)
+                except Exception as e:
+                    print(f"Failed to delete {file_name} due to {e}... Skipping")
+        elif source_file_name_match_type == "exact_match":
+            sftp.remove(source_full_path)
+    except ExitCodeException as e:
+        logger.error(e)
```

Comment:
Should be e.message

Reply:
Similar to above comment. (e) and (e.message) essentially log the same output